yapapi/executor/utils.py

```diff
@@ -27,21 +27,23 @@ def __init__(self, wrapped: Callable, event_loop: Optional[asyncio.AbstractEvent
         self._args_buffer = asyncio.Queue()
         self._task = None
         self._loop = event_loop or asyncio.get_event_loop()
+        self._task = self._loop.create_task(self._worker())
 
     async def _worker(self) -> None:
         while True:
             (args, kwargs) = await self._args_buffer.get()
             self._wrapped(*args, **kwargs)
             self._args_buffer.task_done()
 
-    async def __aenter__(self):
-        self._task = self._loop.create_task(self._worker())
-
-    async def __aexit__(self, exc_type, exc_val, exc_tb):
+    async def _delete(self) -> None:
         await self._args_buffer.join()
         if self._task:
             self._task.cancel()
-        self._task = None
+            await asyncio.gather(self._task, return_exceptions=True)
+            self._task = None
+
+    def __del__(self):
+        self._loop.run_until_complete(self._delete())
```

Comment:
I'm using `__del__()` instead of `AsyncContextManager`'s `__aexit__()` so that an `AsyncWrapper` instance is not closed in `Executor.__aexit__()` but later. Otherwise, event emitter would not be able to emit the `ShutdownFinished` event.